cmd/push-notification-server/main.go

```diff
@@ -0,0 +1,280 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"database/sql"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"path"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/status-im/status-go/appdatabase"
+	"github.com/status-im/status-go/common/dbsetup"
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/logutils"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/pkg/sentry"
+	"github.com/status-im/status-go/pkg/version"
+	"github.com/status-im/status-go/protocol"
+	"github.com/status-im/status-go/protocol/pushnotificationserver"
+	mailserversDB "github.com/status-im/status-go/services/mailservers"
+	"github.com/status-im/status-go/services/personal"
+	"github.com/status-im/status-go/timesource"
+	"github.com/status-im/status-go/waku/types"
+	"github.com/status-im/status-go/wakuv2"
+	wakuv2common "github.com/status-im/status-go/wakuv2/common"
+	"github.com/status-im/status-go/walletdatabase"
+)
+
+var (
+	gorushURL       = flag.String("gorush-url", pushnotificationserver.DefaultGorushURL, "gorush server URL")
+	dataDir         = flag.String("data-dir", "", "data directory")
+	identity        = flag.String("identity", "", "Hex-encoded private key to use. When empty, an ephemeral key will be used")
+	sentryEnabled   = flag.Bool("sentry", false, "Enable sentry panic reporting")
+	logLevel        = flag.String("log-level", "INFO", `Log level, one of: "ERROR", "WARN", "INFO", "DEBUG"`)
+	logNoColors     = flag.Bool("log-no-color", false, "Disables log colors")
+	wakuFleet       = flag.String("waku-fleet", "status.prod", "Waku fleet to use")
+	wakuFleetConfig = flag.String("waku-fleet-config", "", "path to Waku fleet config file")
```

Comment:
Is this exclusive or does it mean other fleets would be discovered eventually?

Reply:
@jakubgs there're only static node, so no other fleets are discovered.

When running tests, we use this local fleet:
https://github.com/status-im/status-go/blob/4f33989d98adddeebd626590c66a3e21c7ca5625/tests-functional/waku_configs/wakufleetconfig.json#L4-L15

Reply:
Nice.